### PR TITLE
ci: Additionally publish public image to ghcr.io

### PR DESCRIPTION
**Pull request by viccuad**
Push the public image also to ghcr.io, under https://github.com/orgs/rancher/packages.

I haven't tested the PR, as I wouldn't progress through the previous steps from my fork.

This is triggered by the desire to consume this image as part of Kubewarden, [here](https://github.com/kubewarden/helm-charts/blob/main/charts/kubewarden-controller/values.yaml#L160). Kubewarden charts follow the convention on having a value `  cattle.systemDefaultRegistry`, which we already have set as `ghcr.io` for the rest of images. Hence, continuing on serving from 1 repository alone and keeping the Rancher convention would be of help to us.

### CODE REVIEW

File: .github/workflows/head-build.yml
Comment (mallardduck) [inline .github/workflows/head-build.yml:81]: This workflow only covers the "head" builds and not stable releases. For your usage I suspect you should only target using the stable tagged releases. Also this will only cover getting the "rolling" tag to GHCR - and we'll also want the static tag mirrored as well.

**Comment by mallardduck** [comment]
@viccuad - This is good now I think. However this may want to wait until after the RFC is fully accepted. Granted I'm biased because I created the project and RFC to start using `kuberlr` for `rancher/shell`...but I do think the odds that it's accepted are pretty good. However if that were to not happen then this project and image wouldn't be maintained further - as the current status is essentially "experimental but stable" for lack of better term.

Once the RFC is approved by RAT, then it will be more certain as a fully supported solution though. However until that happens the future of this image/project is "uncertain" - since if not accepted we'd simply archive this repo most likely.

That in mind, I will feel better if we hold off on merging this for now. To prevent publishing more images than the existing test images on DockerHub today.

**Comment by viccuad** [comment]
Many thanks for the fixes!
I'm ok waiting for the RFC resolution, seems fair.

**Comment by kkaempf** [comment]
@flavio - what's the status of the RFC mentioned by @mallardduck above ?

**Comment by flavio** [comment]
The RFC has been approved on Oct 28

**Comment by mallardduck** [comment]
My team will begin to plan "official launch" of this project and a timeline integration into rancher. Once I have finalized those details I will merge this PR, then prepare necessary changes for release branches and tag the first releases.

**Review by mallardduck — APPROVED** [review]

**Comment by mallardduck** [comment]
BTW, our official target for launch is Rancher 2.11 (in March). However I am anticipating the ability to backport these changes to prior versions of Rancher as well. To that end, we're tagging `main` here against the `r/r@main` branch, and targeting it to be version 4.0.

Doing this may seem odd at first, but it allows for us to publish older versions of this image that target older versions of Rancher.

**Pull request merged**
